app/src/main/java/xyz/jhughes/laundry/activities/MachineActivity.java

```diff
@@ -68,14 +73,43 @@ protected void onCreate(Bundle savedInstanceState) {
 
         setUpViewPager();
         setScreenName(Constants.getApiLocation(currentRoom));
-        updateFilteringTextView();
+        updateFilteringWarning();
     }
 
-    private void updateFilteringTextView() {
+    private void updateFilteringWarning() {
         final SharedPreferences p = SharedPrefsHelper.getSharedPrefs(MachineActivity.this);
         boolean filtering = p.getBoolean("filter", false);
-        findViewById(R.id.machine_activity_filtering_textview).setVisibility(
-                filtering ? View.VISIBLE : View.GONE);
+
+        if(filterWarningBar == null) {
+            filterWarningBar = Snackbar.make(mMainContent, "Only showing available machines.", Snackbar.LENGTH_INDEFINITE)
+                    .setAction("Show all", new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            setFilter(false);
+                        }
+                    }).setCallback(new Snackbar.Callback() {
```

Comment:
Apparently this was deprecated even in the few hours since I used it. (I honestly don't remember seeign the deprecation when I coded it. Though I'm sure it was there and I just missed it. 

Note to self: replace with this

```
.addCallback(new BaseTransientBottomBar.BaseCallback<Snackbar>() {
                        @Override
                        public void onDismissed(Snackbar transientBottomBar, int event) {
                            super.onDismissed(transientBottomBar, event);
                        }
                    })
```